tor/controller.go

```diff
@@ -176,6 +181,15 @@ func (c *Controller) authenticate() error {
 			"%v", err)
 	}
 
+	// If a controlPassword is provided, we try to authenticate using this.
```

Comment:
Since we're introducing a new authentication method, I think it'd be better to refactor this into the following:

```
func (c *Controller) authenticate() error {
    // If a control password was provided, we'll assume the caller intends to use HashedControlPassword authentication.
    if len(c.controlPass) > 0 {
        return c.authenticateWithHashedPass()
    }
    return c.authenticateWithSafeCookie()
}
```